### PR TITLE
Fixes xml token case and returns an explicit error on an unexpected status code

### PR DESCRIPTION
**Pull request by chmllr**
@fiorix as discussed here is my PR, with tests for the camel case. Note one of the existing tests fails currently (also on master).

Also please take a look to my fix of the import path of `wsdl`, I'm not totally sure about that part.

Thank you!

### CODE REVIEW

File: soap/client.go [hunk 1]
```diff
@@ -81,6 +83,10 @@ func (c *Client) RoundTrip(in, out Message) error {
 		return err
 	}
 	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
```
Comment (fiorix) [inline soap/client.go]: Can you wrap `resp.Body` into an `io.LimitedReader` and only read the first few MBs of the response at most?

File: wsdlgo/encoder.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 	"strings"
 	"text/template"
 
-	"github.com/fiorix/wsdl2go/wsdl"
+	"../wsdl"
```
Comment (fiorix) [inline wsdlgo/encoder.go]: This does not work, please keep the original import path. Go is very opinionated on this.

File: wsdlgo/encoder_test.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 	"path/filepath"
 	"testing"
 
-	"github.com/fiorix/wsdl2go/wsdl"
+	"../wsdl"
```
Comment (fiorix) [inline wsdlgo/encoder_test.go]: Ditto

File: soap/client.go (hunk 1 above)
Comment (chmllr) [inline soap/client.go]: Good idea, thanks!

**Comment by chmllr** [comment]
Ready for the next iteration.

**Comment by fiorix** [comment]
Cool, thanks! 👍

**Comment by chmllr** [comment]
Thank you very much. I appreciate a lot your fast reaction 👍

**Pull request merged**